Network/Mail/SMTP.hs

```diff
@@ -137,8 +137,14 @@ parseResponse conn = do
     (code, bdy) <- readLines
     return (read $ B8.unpack code, B8.unlines bdy)
   where
+    getLines c acc = do
```

Comment:
@parsonsmatt I need some help with this as I am not sure if this is right or should I just stick with `connectionGetLine`. Having to specify max bytes per line seems a little off to me.

Reply:
thinking out loud here:

- This function is replacing `B8.hGetLine :: Handle -> IO ByteString`. [Implementation linked](https://www.stackage.org/haddock/lts-13.25/bytestring-0.10.8.2/src/Data-ByteString.html#hGetLine).
- This function reads from the bytestring until it either finds a `\n` character or the end-of-file. (figuring this out exactly took me a long time, dang the bytestring internals are tough)
- `connectionGetLine` doesn't work in exactly the same way. The `Int` parameter is to prevent DoS attacks, so it'd be a security improvement to have it for sure - no idea what typical line length limit might be safe here :man_shrugging: .
- The bigger problem with `connectionGetLine` is that it throws `isEOFError` exception on end of input. Except, it says "An end of file will be considered as a line terminator too, if line is not empty." - Looking at the source... well, it's [a bit confusing, but I think I've worked out what is going on](https://github.com/vincenthz/hs-connection/issues/42).

I think we're fine with `connectionGetLine`, we just need to pick a suitable line length.

Reply:
I did a quick Google and [got this StackOverflow post re line lengths in mail](https://stackoverflow.com/questions/11794698/max-line-length-in-mail).

I think 1000 is a reasonable pick. The standard says lines `MUST` be no more than 998 characters.